### ISSUE TITLE
How to increase specificity of atomic css classnames when passed to component?

### ISSUE THREAD

**Issue by xsjcTony** [ant-design/ant-design#46707]
### Reproduction link

[https://stackblitz.com/edit/vitejs-vite-lvffpm?file=src%2FApp.tsx](https://stackblitz.com/edit/vitejs-vite-lvffpm?file=src%2FApp.tsx)

### Steps to reproduce

Open developer tool

### What is expected?

Is there a way to let the atomic CSS classname override the ant design's style in this scenario? Due to the `>` selector, it has nothing to do whether to use `:where()` or not. And also I don't want to use `important:` modifier for those atomic classnames

### What is actually happening?

The classname `p-4` and `rounded-md` have no effect due to `.ant-input-affix-wrapper >input.ant-input` selector has higher specificity when `prefix` or `suffix` is defined.
![Snipaste_2023-12-30_16-36-13](https://github.com/ant-design/ant-design/assets/45784210/2f162616-e633-4e8c-b97c-c8881cf7ef07)
![Snipaste_2023-12-30_16-35-55](https://github.com/ant-design/ant-design/assets/45784210/2e0a33ba-2dd1-4f69-a755-135d412e0ac4)

### Additional comment
Chinese response is acceptable.

| Environment | Info |
| --- | --- |
| antd | 5.12.5 |
| React | 18.2.0 |
| System | Windows 11 |
| Browser | Chrome 120 |

**Comment by yoyo837** [comment]
FYI: https://ant.design/docs/react/faq#css-in-js-css-priority-conflict-with-tailwindcss

**Comment by xsjcTony** [comment]
> FYI: https://ant.design/docs/react/faq#css-in-js-css-priority-conflict-with-tailwindcss

Yeah I've had a look into this before I raise this but I don't think it's the same as my issue.

It may worth a try of important setting but I'm not sure whether it's available in UnoCSS. I'll have a look and come back. Although I don0t really think '#root .p-3' makes sense

**Comment by xsjcTony** [comment]
No unfortunately UnoCSS does not currently have this feature :(

**Comment by yoyo837** [comment]
https://github.com/ant-design/ant-design/issues/38794#issuecomment-1328262525

https://stackblitz.com/edit/nextjs-c2ufm8?file=styles%2FHome.module.css,pages%2Findex.js,pages%2F_app.js

Did the above help you?

**Comment by xsjcTony** [comment]
@yoyo837 That doesn't. Have a look at this. It's going to be overridden because the `input` element is in are wrapper with antd classname, so it's `.hash.class1 > .class2` vs `.atomic`:
 https://stackblitz.com/edit/nextjs-si1csf?file=pages%2Findex.js

**Comment by zombieJ** [comment]
In my view, CSS selector with `#root` selector for atomic className is the current answer.

In early v5 alpha version, I try to make antd style with `@layer` scope to reduce antd style priority, but found that `@layer` is too low that get bad experience more than it get.

Since it's a not certain issue with antd, let's move this to the discussion instead.
